jerry-core/ecma/base/ecma-helpers.h

```diff
@@ -52,6 +52,33 @@
  */
 #define ECMA_SET_POINTER(field, non_compressed_pointer) MEM_CP_SET_POINTER (field, non_compressed_pointer)
 
+/**
+ * Convert ecma-string's contents to a cesu-8 string and put it into a buffer.
+ */
+#define ECMA_STRING_TO_UTF8_STRING(ecma_str_ptr, utf8_ptr) \
+  lit_utf8_size_t utf8_ptr ## _size; \
+  bool utf8_ptr ## _is_ascii; \
+  const lit_utf8_byte_t *utf8_ptr = ecma_string_raw_chars (ecma_str_ptr, &utf8_ptr ## _size, &utf8_ptr ## _is_ascii); \
+  bool utf8_ptr ## must_be_freed = false; \
```

Comment:
I tried grepping around in the code base but could not find any occurrence where the `_completion` variable declared by `ECMA_TRY_CATCH` would be used outside the macro or its pair `ECMA_FINALIZE`. So, yes, those macros also declare and use variables but those variables are considered "local" and are not used in the code that expands those macros.